### PR TITLE
Independence Day: Add missing countries

### LINKED ISSUE #2790: IndependenceDay: Missing a few countries
Hello, I noticed a few countries are missing in the independence_day.json file as well as no IA's are provided for those countries while searching.

I would like to contribute by adding those infos but I need guidance as it'll be my first contribution in IA. Will updating the independence_day.json file with the info be enough?? or do I need to add something else as well??

------
IA Page:  http://duck.co/ia/view/independence_day 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): ~~@jarmokivekas~~ @RCMainak

### PR DESCRIPTION
**Pull request by t2013anurag**
Fixes #2790

Fixes references : http://www.state.gov/misc/list/index.htm
and https://en.wikipedia.org/wiki/List_of_countries_and_dependencies_and_their_capitals_in_native_languages

---
IA Page : https://duck.co/ia/view/independence_day
Maintainer: @jarmokivekas

### CODE REVIEW

**Comment by daxtheduck** [comment]
@t2013anurag Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/independence_day/country_aliases.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
     "arab emirates": "united arab emirates",
     "barbuda": "antigua and barbuda",
     "bosnia": "bosnia and herzegovina",
+    "ivory coast" : "Côte d'Ivoire",
```
Comment (moollaza) [inline share/goodie/independence_day/country_aliases.json]: nit: it looks like this file is alpha-sorted by the keys so this one should be a little lower on the list, after `herzegovina`

**Comment by moollaza** [comment]
@t2013anurag thanks for updating this!

@jarmokivekas you're the maintainer of this IA. Please let us know if you approve that changes!

**Comment by t2013anurag** [comment]
@moollaza : Thanks for pointing out.

**Comment by t2013anurag** [comment]
@moollaza : I have updated the list alphabetically. Thanks for pointing out the bug.

**Comment by RCMainak** [comment]
@t2013anurag bro, if you have done a bit of research before submitting the patch it would have saved my ass from typing in few lines. Anyways, thank you for letting me know the process :+1: 

Improved version of the fix can be viewed here : #2792 
Happy Holi everyone :smile:

**Comment by t2013anurag** [comment]
@moollaza  - I have merged both #2791 and #2792 as assigned to me. Can you please request maintainer input as I guess he is not active from last some days.

Thanks @RCMainak  for your contribution.
Great work :)

**Comment by RCMainak** [comment]
@t2013anurag you just made things more complex :tongue: my PR had everything there. You did not need to megre it with your own PR. Can't you take a few moment to read what is going on before you act!!! 
Anyways, waiting for maintainers input... thank you for your support

**Comment by jarmokivekas** [comment]
I'm the original author of this IA. I'm quite busy with other things, and I'm quite unlikely to be able to put in the time and effort to thoroughly review pull requests. If the community in general agrees on changes, go ahead and commit them.

**Comment by mintsoft** [comment]
Will do @jarmokivekas Thanks for the input

**Comment by RCMainak** [comment]
@moollaza @mintsoft @jarmokivekas  If on one have any problem I would like to take up the opportunity to maintain this IA. As I'm new to the community please redirect me to place where I can do it officially.

**Comment by jarmokivekas** [comment]
@RCMainak I'm not up-to-date on how maintainers are assigned, but I'm glad to relinquish my status as maintainer since unfortunately I have other priorities.

**Comment by moollaza** [comment]
> I'm not up-to-date on how maintainers are assigned, but I'm glad to relinquish my status as maintainer since unfortunately I have other priorities.

@jarmokivekas thanks for letting us know! I'll remove you as maintainer.


> If on one have any problem I would like to take up the opportunity to maintain this IA. As I'm new to the community please redirect me to place where I can do it officially.

@RCMainak thanks for offering, I will set you as the maintainer

@t2013anurag thanks for combinig the PRs

File: share/goodie/independence_day/independence_days.json [hunk 2]
```diff
@@ -80,14 +83,14 @@
     "jordan":  [{"date": "May 25th", "year": "1946"}],
     "kazakhstan":  [{"date": "December 16th", "year": "1991"}],
     "kenya":  [{"date": "December 12th", "year": "1963"}],
+    "Kiribati":  [{"date": "July 12th", "year": "1979"}],
```
Comment (moollaza) [inline share/goodie/independence_day/independence_days.json]: lowercase

File: share/goodie/independence_day/country_aliases.json [hunk 3]
```diff
@@ -4,10 +4,12 @@
     "barbuda": "antigua and barbuda",
     "bosnia": "bosnia and herzegovina",
     "democratic republic of congo": "democratic republic of the congo",
+    "east timor" : "Timor-Leste",
```
Comment (moollaza) [inline share/goodie/independence_day/country_aliases.json]: add tests for these -- they're not working currently

Comment (moollaza) [inline share/goodie/independence_day/country_aliases.json]: You need to flip these around. The right hand is the lookup key into the hash below

**Comment by moollaza** [comment]
@t2013anurag please add tests for the new queries. It looks like there's still some work to be done as the aliases are backwards

**Comment by RCMainak** [comment]
@t2013anurag Are you going to fix the issues??? Please kindly respond even if it is not possible for you to make the changes or if you're working on the issues...

**Comment by tagawa** [comment]
@t2013anurag, thanks again for fixing the Independence Day IA! You’ve been doing a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in country topics.

When you're ready to move on, please:

- Create a new Instant Answer for searches you do related to country topics

OR

- Test live country topic Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

**Comment by t2013anurag** [comment]
@RCMainak  Sorry lately I've been occupied with a lot of work. Will do it asap !

**Comment by t2013anurag** [comment]
@RCMainak I've updated the tests and all the changes have been done

**Comment by tagawa** [comment]
Thanks for the quick updates @t2013anurag - I didn't mean to pressure you!

If you'd like to help in other areas, you could:

* Create a new Instant Answer for searches you do related to country topics

OR

* Test live country topic Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this page: https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers

**Comment by RCMainak** [comment]
@t2013anurag Thank you for your contribution :smiley:

**Pull request merged**